pajbot/managers/irc.py

```diff
@@ -1,63 +1,155 @@
 import logging
+import ssl
 
-from pajbot.managers.connection import ConnectionManager
+import socket
+from irc.client import InvalidCharacters, MessageTooLong, ServerConnection, ServerNotConnectedError
+from irc.connection import Factory
+from ratelimiter import RateLimiter
+
+from pajbot.managers.schedule import ScheduleManager
 
 log = logging.getLogger(__name__)
 
 
-def do_nothing(_c, _e):
-    pass
+class Connection(ServerConnection):
+    """
+    We override the default irc.clientServerConnection because want to be able
+    to send strings that are 2048(?) bytes long. This is not in accordance with the IRC
+    standards, but it's the limit the Twitch IRC servers use.
+    """
+
+    def send_raw(self, string):
+        """Send raw string to the server.
+
+        The string will be padded with appropriate CR LF.
+        """
+        # The string should not contain any carriage return other than the
+        # one added here.
+        if "\n" in string or "\r" in string:
+            raise InvalidCharacters("CR/LF not allowed in IRC commands")
+        bytes = string.encode("utf-8") + b"\r\n"
+        # According to the RFC http://tools.ietf.org/html/rfc2812#page-6,
+        # clients should not transmit more than 512 bytes.
+        # However, Twitch have raised that limit to 2048 in their servers.
+        if len(bytes) > 2048:
+            raise MessageTooLong("Messages limited to 2048 bytes including CR/LF")
+        if self.socket is None:
+            raise ServerNotConnectedError("Not connected.")
+        sender = getattr(self.socket, "write", self.socket.send)
+        try:
+            sender(bytes)
+        except socket.error:
+            # Ouch!
+            self.disconnect("Connection reset by peer.")
 
 
 class IRCManager:
```

Comment:
Just generally, I combined IRCManager and ConnectionManager to make it more understandable what's actually going on. The entire IRC connection handling is now in this one class.